test/rekt/features/kafkasource/data_plane.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafkasource
+
+import (
+	"context"
+	"os"
+	"time"
+
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	assert "knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+
+	sourcesv1beta1 "knative.dev/eventing-kafka/pkg/apis/sources/v1beta1"
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkacat"
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkasource"
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkatopic"
+)
+
+var (
+	test_mt_source = os.Getenv("TEST_MT_SOURCE")
+)
+
+// DataPlaneDelivery returns a feature testing if the event sent to Kafka is received by the sink
+// via KafkaSource installed before sending events to Kafka
+func DataPlaneDelivery(name string, ksopts []kafkasource.CfgFn, kcopts []kafkacat.CfgFn, matchers cetest.EventMatcher, intermediary bool) *feature.Feature {
+	f := feature.NewFeatureNamed("Delivery")
+
+	// Setup sink
+	sinkName := feature.MakeRandomK8sName("sink")
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiver))
+
+	ksopts = append(ksopts, kafkasource.WithSink(&duckv1.KReference{
+		Kind:       "Service",
+		Name:       sinkName,
+		APIVersion: "v1",
+	}, ""))
+
+	// Setup topic
+	topicName := feature.MakeRandomK8sName("topic") // A k8s name is also a valid topic name.
+	f.Setup("install kafka topic", kafkatopic.Install(topicName))
+
+	ksopts = append(ksopts, kafkasource.WithTopics([]string{topicName}))
+	kcopts = append(kcopts, kafkacat.WithTopic(topicName))
+
+	// Setup source
+	f.Setup("install a kafka source", kafkasource.Install(name, ksopts...))
+
+	f.Requirement("kafka topic must be ready", kafkatopic.IsReady(topicName))
+	f.Requirement("kafka source must be ready", kafkasource.IsReady(name))
```

Comment:
it feels more like an assertion to me.

Reply:
The same source of confusion as in the other places: I also expressed confusion about using Requirement in knative-sandbox/reconciler-test#106 (comment) . This single check we're commenting on could easily be wrapped/verified in the Setup step a few lines above. I believe this will continue to be a source of confusion until the docs precisely specify its purpose.